README.md

```diff
@@ -26,24 +26,80 @@ This library provides functionality similar to [ipfs-unixfs-importer][], but it
 
    Library removes indirection by taking approach similar to [multiformats][] library. Instead of passing chunker and layout config options, you pass chunker / layout / encoder interface implementations.
 
-### Importing a file
+### Usage
 
-To encode a file you simply create file importer and write conten
+You can encode a file as follows
 
 ```js
-import * as UnixFSFile from "@ipld/unixfs/src/file.js"
+import * as UnixFS from "@ipld/unixfs"
+
+// Create a web `TransformStream` with additional filesystem specific interface
+// that allows encoding files and directories into `writable` end and reading
+// IPLD blocks from `readable` end.
+const archive = UnixFS.create()
+
+// Create file writer that can be used to encode UnixFS file.
+const file = UnixFS.createFileWriter(fs)
+// write some content
+file.write(new TextEncoder().encode("hello world"))
+// Finalize file by closing it.
+const { cid } = await file.close()
+
+// close the archive to close underlying block stream.
+archive.close()
 
-const encodeFile = (blob) => {
-  const { writer, ...importer } = FileImporter.createImporter()
-  // Collects data coming out of ReadableStream into array
-  const blocks = collect(importer.blocks)
+// We could encode all this as car file
+encodeCAR({ roots: [cid], blocks: archive.blocks })
+```
+
+If your runtime provides [`TransforStream`][] or a [`WritableStream`][] APIs you can create filesystem writer from the `WritableStream<Block>` directly (this will allow you to control exactly how `readable`/`writable` streams deal with backpressuer)
```

Comment:
```suggestion
If your runtime provides [`TransformStream`][] or a [`WritableStream`][] APIs you can create filesystem writer from the `WritableStream<Block>` directly (this will allow you to control exactly how `readable`/`writable` streams deal with back pressure)
```